### PR TITLE
Feature xmlexport

### PR DESCRIPTION
**Pull request by Emiel94**
One question: is this git ignore correct?

### CODE REVIEW

File: contextproject-ms3/src/main/java/contextproject/formats/XmlExport.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import java.io.FileOutputStream;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import contextproject.helpers.StackTrace;
+import contextproject.loaders.FolderLoader;
+import contextproject.loaders.LibraryLoader;
+import contextproject.models.Key;
+import contextproject.models.Playlist;
```
Comment (Roy-Graafmans) [inline contextproject-ms3/src/main/java/contextproject/formats/XmlExport.java]: wrong input order

File: contextproject-ms3/src/main/java/contextproject/formats/XmlExport.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * Class to export data to a XML file.
+ *
+ */
+public class XmlExport {
+
+  private static Logger log = LogManager.getLogger(LibraryLoader.class.getName());
+  private File file;
+  private Playlist playlist;
+
+  public XmlExport(String location, Playlist pl) {
```
Comment (Roy-Graafmans) [inline contextproject-ms3/src/main/java/contextproject/formats/XmlExport.java]: need javadoc

File: contextproject-ms3/src/main/java/contextproject/models/BeatGrid.java [hunk 3]
```diff
@@ -67,5 +74,36 @@ public BeatRange getIntro() {
...
+    outro = outroRange;
+  }
+  
+  public void setFirstBeat(long fb) {
+    firstBeat = fb;
+  }
+  
+  public void setTimePerBeat(double tpb) {
+    timePerBeat = tpb;
+  }
```
Comment (Roy-Graafmans) [inline contextproject-ms3/src/main/java/contextproject/models/BeatGrid.java]: all getters and setters need javadoc

File: contextproject-ms3/src/main/java/contextproject/models/Track.java [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
+import java.util.Hashtable;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 import com.mpatric.mp3agic.InvalidDataException;
 import com.mpatric.mp3agic.Mp3File;
 import com.mpatric.mp3agic.UnsupportedTagException;
 
 import contextproject.helpers.StackTrace;
```
Comment (Roy-Graafmans) [inline contextproject-ms3/src/main/java/contextproject/models/Track.java]: wrong input order again

File: contextproject-ms3/.gitignore [hunk 5]
```diff
@@ -1,2 +1,3 @@
 /target/
 /playlists/
+/src/test/resources/test.xml
```
Comment (daveystruijk) [inline contextproject-ms3/.gitignore]: Remove this

File: contextproject-ms3/src/test/resources/loadTest/test.xml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<java version="1.8.0_25" class="java.beans.XMLDecoder">
+ <object class="contextproject.models.Playlist"/>
+</java>
```
Comment (daveystruijk) [inline contextproject-ms3/src/test/resources/loadTest/test.xml]: This should be deleted after testing

**Pull request merged**